source/DSCResources/DSC_CMServiceConnectionPoint/DSC_CMServiceConnectionPoint.psm1

```diff
@@ -0,0 +1,227 @@
+        [ValidateSet('Present','Absent')]
+        [String]
+        $Ensure = 'Present'
+    )
+
+    Import-ConfigMgrPowerShellModule -SiteCode $SiteCode
+    Set-Location -Path "$($SiteCode):\"
+    $state = Get-TargetResource -SiteCode $SiteCode -SiteServerName $SiteServerName
+
+    try
+    {
+        if ($Ensure -eq 'Present')
+        {
+            if ($state.Ensure -eq 'Absent')
+            {
+                Write-Verbose -Message ($script:localizedData.AddSCPRole -f $SiteServerName)
+                Add-CMServiceConnectionPoint -SiteSystemServerName $SiteServerName -SiteCode $SiteCode -Mode $Mode
+            }
+
+            elseif ($state.Mode -ne $Mode)
+            {
+                Write-Verbose -Message ($script:localizedData.SettingValue -f $Mode)
+                Set-CMServiceConnectionPoint -SiteSystemServerName $SiteServerName -SiteCode $SiteCode -Mode $Mode
+            }
+        }
+
+        elseif ($state.Ensure -eq 'Present')
+        {
+            Write-Verbose -Message ($script:localizedData.RemoveSCPRole -f $SiteServerName)
+            Remove-CMServiceConnectionPoint -SiteSystemServerName $SiteServerName -SiteCode $SiteCode
+        }
+    }
+    catch
+    {
+        throw $_
+    }
+    finally
+    {
+        Set-Location -Path "$env:temp"
+    }
+}
+
+<#
+    .SYNOPSIS
+        This will test the desired state.
+
+    .PARAMETER SiteCode
+        Specifies a site code for the Configuration Manager site.
+
+    .PARAMETER SiteServerName
+        Specifies the Site Server to install or configure the role on.
+
+    .PARAMETER Mode
+        Specifies a mode for the service connection point. The acceptable values are Online and Offline.
+
+    .PARAMETER Ensure
+        Specifies whether the fallback status point is present or absent.
+
+    .Notes
+        This role must only be installed on top-level site of the hierarchy.
+        This role can only be installed on CAS or PRI.
+#>
+
+function Test-TargetResource
+{
+    [CmdletBinding()]
+    [OutputType([System.Boolean])]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [String]
+        $SiteCode,
+
+        [Parameter(Mandatory = $true)]
+        [String]
+        $SiteServerName,
+
+        [Parameter()]
+        [ValidateSet('Online', 'Offline')]
+        [String]
+        $Mode = 'Online',
+
+        [Parameter()]
+        [ValidateSet('Present','Absent')]
+        [String]
+        $Ensure = 'Present'
+    )
+
+    Import-ConfigMgrPowerShellModule -SiteCode $SiteCode
+    Set-Location -Path "$($SiteCode):\"
+    $state = Get-TargetResource -SiteCode $SiteCode -SiteServerName $SiteServerName
+    $result = $true
+
+    if ($Ensure -eq 'Present')
+    {
+        if ($state.Ensure -eq 'Absent')
+        {
+            Write-Verbose -Message ($script:localizedData.SCPNotInstalled -f $SiteServerName)
+            $result = $false
+        }
+
+        if (([string]::IsNullOrEmpty($state.mode)) -or ($state.Mode -ne $Mode))
```

Comment:
([string]::IsNullOrEmpty($state.mode) - This can this happen?

Reply:
Yeah, if absent, mode can be null.